include/deal.II/base/exception_macros.h

```diff
@@ -616,14 +616,14 @@
         }                                                                     \
       while (false)
 #  else
-#    define Assert(cond, exc)  \
-      do                       \
-        {                      \
-          if constexpr (false) \
-            if (!(cond))       \
-              {                \
-              }                \
-        }                      \
+#    define Assert(cond, exc) \
+      do                      \
+        {                     \
+          if (false)          \
```

Comment:
Huh, that's wild!...

Reply:
I think that this variant should not print unused variable warnings with any configuration - we'll see how well it turns out once we do a full CI run.